### PR TITLE
Use numpy instead of torch to get random seed

### PR DESCRIPTION
**Pull request by YuanTingHsieh**
Signed-off-by: YuanTingHsieh <yuantinghsieh@gmail.com>

Fixes # .

### Description
For the original way, if we load the model using TorchScript VS load the model using checkpoints.
It will affect the value generated by torch.empty().random_.
If we change to this new way, then this seed will be the same for loading model using either TorchScript or checkpoints.

Note that, even we don't make this change.
The original code is already repeatable. (run several times yield same result)
So this change is not necessary.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by wyli** [comment]
Hi @YuanTingHsieh the default multiprocess random seeds are from here: https://github.com/Project-MONAI/MONAI/blob/067875b847605ac97ad49569875288a82b101320/monai/data/utils.py#L226-L227  the current implementation ensures the same seed when `num_workers` is  0 or 1. I don't completely understand your use case, if really necessary, we could add an option `np_pt_rnd` in DataLoader

**Comment by YuanTingHsieh** [comment]
Hi @wyli 

The use case is for verification of the exported model VS trained model using the same pipeline.
If the pipeline contains random transforms and we use MONAI's set_determinism

we do

```
composed_trans = Compose([LoadImaged, SomeRandomTransform])
model = torch.jit.load('model.ts')
data_loader = DataLoader(composed_trans)

# doing some validation
```

VS

```
composed_trans = Compose([LoadImaged, SomeRandomTransform])
model = UNet()
model.load_state_dict(torch.load('model.pt'))
data_loader = DataLoader(composed_trans)

# doing some validation
```

If we set the seed using numpy, then with num_workers == 0 we can guarantee these two results are the same.
But again, I am not sure if we need to make this happen.
@Nic-Ma  can add some comments.

**Comment by wyli** [comment]
ok, in your code example, the cases mainly differ in the model initiailisation 
method 1: 
```model = torch.jit.load('model.ts')```
and method2:
```model = UNet(); model.load_state_dict(torch.load('model.pt'))```

but I don't see why they affect the torch random states, could you explain more?

**Comment by YuanTingHsieh** [comment]
Yeah I don't understand why model init would affect random state either.
But I observe that is what is happening.

You can run the following two codes and check.

```
import torch
import numpy as np

from monai.networks.nets import Unet
from monai.utils import set_determinism

set_determinism(seed=0)
model = Unet(
          dimensions=3,
          in_channels=2,
          out_channels=2,
          channels=[16, 32, 64, 128, 256],
          strides=[2, 2, 2, 2],
          num_res_units=2,
          norm='batch'
        )
model.load_state_dict(torch.load('./models/model.pt'))

torch_seed = torch.empty((), dtype=torch.int32).random_(generator=None).item()
numpy_seed = np.random.randint(np.iinfo(np.int32).max + 1)
print(f'Torch seed is {torch_seed}')
print(f'Numpy seed is {numpy_seed}')

```

On my ubuntu this yields:
Torch seed is 971764478
Numpy seed is 209652396

VS

```
import torch
import numpy as np

from monai.networks.nets import Unet
from monai.utils import set_determinism

set_determinism(seed=0)
model = torch.jit.load('./models/model.ts')

torch_seed = torch.empty((), dtype=torch.int32).random_(generator=None).item()
numpy_seed = np.random.randint(np.iinfo(np.int32).max + 1)
print(f'Torch seed is {torch_seed}')
print(f'Numpy seed is {numpy_seed}')

```

This yields:
Torch seed is 209652396
Numpy seed is 209652396

```

**Comment by wyli** [comment]
thanks for the detailed investigation, I think we couldn't fully resolve this unless we change this pytorch impl., correct? https://github.com/pytorch/pytorch/blob/1a92802bde28286f850b415bfd985515565740c3/torch/utils/data/dataloader.py#L498

**Comment by YuanTingHsieh** [comment]
@wyli I think you are right.
@madil90 @Nic-Ma You can see the discussion

So we can just close this PR then?

**Comment by Nic-Ma** [comment]
Hi @YuanTingHsieh ,

Yes, maybe you can close this PR now, but we need to keep it in mind and try to figure it out someday, or at least make sure we can explain to others why the TorchScript result with the random operation is different.
Thanks very much for your deep analysis, now it's much clear for us.

Thanks.
